web/src/lottie-player.ts

```diff
@@ -0,0 +1,577 @@
+    await this._update();
+    this._render();
+  }
+
+  /**
+   * Destroy animation and lottie-player element.
+   * @since 1.0
+   */
+  public destroy(): void {
+    if (!this.TVG) {
+      return;
+    }
+
+    this.TVG = null;
+    this.currentState = PlayerState.Destroyed;
+
+    if (this._observer) {
+      this._observer.disconnect();
+      this._observer = undefined;
+    }
+    
+    this.dispatchEvent(new CustomEvent(PlayerEvent.Destroyed));
+    this.remove();
+  }
+
+  /**
+   * Sets the repeating of the animation.
+   * @param value Whether to enable repeating. Boolean true enables repeating.
+   * @since 1.0
+   */
+  public setLooping(value: boolean): void {
+    if (!this.TVG) {
+      return;
+    }
+
+    this.loop = value;
+  }
+
+  /**
+   * Animation play direction.
+   * @param value Direction values. (1: forward, -1: backward)
+   * @since 1.0
+   */
+  public setDirection(value: number): void {
+    if (!this.TVG) {
+      return;
+    }
+
+    this.direction = value;
+  }
+
+  /**
+   * Set animation play speed.
+   * @param value Playback speed. (any positive number)
+   * @since 1.0
+   */
+  public setSpeed(value: number): void {
+    if (!this.TVG) {
+      return;
+    }
+
+    this.speed = value;
+  }
+
+  /**
+   * Set a background color. (default: 0x00000000)
+   * @param value Hex(#fff) or string(red) of background color
+   * @since 1.0
+   */
+  public setBgColor(value: string): void {
+    if (!this.TVG) {
+      return;
+    }
+
+    this.canvas!.style.backgroundColor = value;
+  }
+
+  /**
+   * Save current animation to other file type
+   * @param target File type to export
+   * @since 1.0
+   */
+  public async save(target: ExportableType): Promise<void> {
+    if (!this.TVG || !this.src) {
+      return;
+    }
+
+    const bytes = await _parseSrc(this.src);
+    
+    switch (target) {
+      case ExportableType.GIF:
+        const isExported = this.TVG.save2Gif(bytes, 'lottie', this.canvas!.width, this.canvas!.height, 30);
+        if (!isExported) {
+          throw new Error('Unable to save a GIF. Error: ', this.TVG.error());
+        }
+
+        const data = _module.FS.readFile('output.gif');
+        if (data.length < 6) {
+          throw new Error("Unable to save the Gif data. The generated file size is invalid.");
+        }
+
+        const blob = new Blob([data], {type: 'application/octet-stream'});
+        const link = document.createElement("a");
+        link.setAttribute('href', URL.createObjectURL(blob));
+        link.setAttribute('download', 'output.gif');
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+        break;
+    }
+  }
+
+  /**
+   * Return thorvg version
+   * @since 1.0
+   */
+  public getVersion(): LibraryVersion {
+    return {
+      THORVG_VERSION,
+    };
+  }
+
+  public render (): TemplateResult {
+    return html`
+      <canvas id="thorvg-canvas" style="width: 100%; height: 100%;" />
+    `;
+  }
+}
```

Comment:
Oh, we don't actually use eslint for operating, I just followed base project. Since we don't adapt specific convention, we can't use it as rule.

Is it better to remove eslint things from project? if it's really confusing.

Reply:
using Eslint is not confusing at all. I suggest keeping it and fixing its errors that help to prevent bugs in the final bundle it makes sure you follow the proper format when other devs import your package and helps you to prevent bugs such us "always using default in a switch case". Of course if you'd like to do it in a different version - go ahead, but I highly recommened fixing eslint errors.